src/data_cell/graph_interface.h

```diff
@@ -24,32 +24,36 @@
 #include "index/index_common_param.h"
 #include "stream_reader.h"
 #include "stream_writer.h"
+#include "typing.h"
 
 namespace vsag {
 
+class GraphInterface;
+using GraphInterfacePtr = std::shared_ptr<GraphInterface>;
+
 class GraphInterface {
 public:
     GraphInterface() = default;
 
     virtual ~GraphInterface() = default;
 
-    static std::shared_ptr<GraphInterface>
-    MakeInstance(const nlohmann::json& json_obj,
+    static GraphInterfacePtr
+    MakeInstance(const JsonType& json_obj,
                  const IndexCommonParam& common_param,
                  bool is_sparse = false);
 
 public:
     virtual void
-    InsertNeighborsById(uint64_t id, const std::vector<uint64_t>& neighbor_ids) = 0;
+    InsertNeighborsById(InnerIdType id, const Vector<InnerIdType>& neighbor_ids) = 0;
 
     virtual uint32_t
-    GetNeighborSize(uint64_t id) = 0;
+    GetNeighborSize(InnerIdType id) const = 0;
 
     virtual void
-    GetNeighbors(uint64_t id, std::vector<uint64_t>& neighbor_ids) = 0;
+    GetNeighbors(InnerIdType id, Vector<InnerIdType>& neighbor_ids) const = 0;
 
     virtual void
-    Prefetch(uint64_t id, uint64_t neighbor_i) = 0;
+    Prefetch(InnerIdType id, uint32_t neighbor_i) = 0;
```

Comment:
`neigbhor_id` change to `uint32_t` for big single node is enough ?

Reply:
use InnerIdType (int32_t)

Reply:
neighbor counts for one node, uint32_t is enough, The InnerIdType is uint32_t